apps/src/studio/studio.js

```diff
@@ -3161,17 +3161,17 @@ var registerHandlers = function (
   matchParam2Val,
   argNames
 ) {
-  var blocks = Blockly.mainBlockSpace.getTopBlocks();
-  for (var x = 0; blocks[x]; x++) {
-    var block = blocks[x];
-    // default title values to '0' for case when there is only one sprite
-    // and no title value is set through a dropdown
-    var titleVal1 = block.getFieldValue(nameParam1) || '0';
-    var titleVal2 = block.getFieldValue(nameParam2) || '0';
+  const blocks = Blockly.mainBlockSpace.getTopBlocks();
+  for (let x = 0; blocks[x]; x++) {
+    const block = blocks[x];
+    // default field values to '0' for case when there is only one sprite
+    // and no field value is set through a dropdown
+    const fieldVal1 = block.getFieldValue(String(nameParam1)) || '0';
```

Comment:
could we instead check if `nameParam1` is `undefined`, and return `'0'` if it is and call `getFieldValue` if it's not?

Reply:
Updated to check if `nameParam1` is not a string.